### PR TITLE
Feature/aws - Run Radish34 POC on AWS EC2

### PR DESCRIPTION
**Pull request by MCLDG**
# Description

I have added a new README, baseline/radish34/README-EC2.md. to describe how to run the POC on an AWS EC2 instance. I had to make some small tweaks to 2-3 files to replace existing hardcoded URL endpoints with configurable endpoints.

## Related Issue

[Issue](https://github.com/ethereum-oasis/baseline/issues/106)

## Motivation and Context

Allows developers to run the Radish34 POC on an AWS EC2 instance instead of using their Macs. Useful for those who do not have Macs or who have Macs without the required capacity (CPU, RAM, etc.)

## How Has This Been Tested

Using the integration testing script: npm run test, and using Firefox to access the Radish home page. 

I've gone through the process twice from beginning to end, but would definitely appreciate someone else sanity checking my README.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

I have added a new README, baseline/radish34/README-EC2.md

- [X ] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X ] All new and existing tests passed.

### CODE REVIEW

**Comment by MCLDG** [comment]
Not sure how to fix the CLA issue - it thinks there are 2 committers, probably because I did not have my email address set in my git config for the earlier commits.

**Comment by skarred14** [comment]
thanks for submitting this PR. For the cla check, you might need to edit commits as an authenticated github user in interactive rebase, and that should solve this issue. Let us know in case this doesnt fix it

**Comment by MCLDG** [comment]
will do @breakpointer , I've pulled the latest master, which unfortunately no longer works on my Mac. I plan to merge my change but need a working baseline (pun, sorry). I'll pull again and retry.

**Comment by kthomas** [comment]
See #125 @MCLDG 

There is a single additional change I needed to make to successfully run radish on EC2 beyond what is in that branch (beyond the obvious of needing Ubuntu dependencies installed before cloning...); that change was updating the `localhost` entry in the `ui` package such that it pointed at my host. Otherwise, the browser attempts to connect to `localhost`.

See `ui/src/contexts/index.js` and `ui/src/components/UserSelection.js` for where this needs to be updated. We could add an environment variable to support that but I just updated it manually for now after building using the branch #125 is based on.

File: docs/basics/glossary.md [hunk 1]
```diff
@@ -88,9 +84,6 @@ MSA: Master Service Agreement is a document/contract issued to mark the beginnin
 
 PO: Purchase order is a document/token issued to begin the ordering process and that needs to maintain linkage to the MSA and thereby the terms agreed upon in the MSA.
 
-
-
-
-
-## 
+\#
```
Comment (skarred14) [inline docs/basics/glossary.md:87]: was this intended?

File: radish34/.gitignore [hunk 2]
```diff
@@ -101,3 +101,4 @@ config/merkle-tree/
 
 # Ignore generated docs
 docs/generated
+.vscode
```
Comment (skarred14) [inline radish34/.gitignore:104]: need a line break here

**Comment by skarred14** [comment]
Looks like this would need to be rebased. Also, please take a look at #125 , this should address the concerns with running radish34 on ubuntu

**Comment by skarred14** [comment]
@MCLDG Recently a PR #125 was merged to address issues of running the radish poc on ubuntu based systems. If that works for you, any concern in closing this in favor of 125?

**Comment by MCLDG** [comment]
Thanks @kthomas , I've tested the changes on AWS and `make`, etc., works correctly.

@skarred14 , there are a few more parts to the demo that I'd like to test before closing this. I would still want to access the UI and try out some of the scenarios. AFAIK we don't have instructions on how to do this (on AWS or other remote platforms), so let me test this. This PR will need rebasing anyway - so I'll either update/rebase it or close it depending on the outcome of my testing.

**Comment by danielnorkin** [comment]
I am having similar issues with running radish34 on Ubuntu in Azure. The UI shows while inside the VM.

image

However, when trying to access the UI from outside of the vm with a hosted dns in Azure, this is what I get.

image

**Comment by skarred14** [comment]
Does this occur even after deleting the local storage on the browser?

**Comment by danielnorkin** [comment]
> Does this occur even after deleting the local storage on the browser?

Yes. Also, note that I didn't make the changes the @kthomas mentioned. What specifically should be changed?

**Comment by kthomas** [comment]
> > Does this occur even after deleting the local storage on the browser?
> 
> Yes. Also, note that I didn't make the changes the @kthomas mentioned. What specifically should be changed?

Well this just sounds like a recipe for disaster! 🤣 

You have to change the hard-coded `localhost` values in the API and websocket urls. Otherwise, they attempt to connect to `localhost`... everywhere...

**Comment by danielnorkin** [comment]
image
graphql works without any changes. we'll continue to poke around

**Comment by kthomas** [comment]
@danielnorkin 

```
ubuntu@ip-172-31-23-163:~/baseline/radish34/ui$ grep -r localhost ./src
./src/serviceWorker.js:  window.location.hostname === 'localhost' ||
./src/serviceWorker.js:    // [::1] is the IPv6 localhost address.
./src/serviceWorker.js:    // 127.0.0.1/8 is considered localhost for IPv4.
./src/serviceWorker.js:        // This is running on localhost. Let's check if a service worker still exists or not.
./src/serviceWorker.js:        // Add some additional logging to localhost, pointing developers to the
./src/serviceWorker.js:        // Is not localhost. Just register service worker
./src/contexts/index.js:let uri = 'localhost:8001/graphql';
./src/components/UserSelection.js:  const apiURL = window.localStorage.getItem('api') || 'localhost:8001/graphql';
./src/components/UserSelection.js:      Org1: { url: 'localhost:8001/graphql', role: 1 },
./src/components/UserSelection.js:      Supplier1: { url: 'localhost:8002/graphql', role: 2 },
./src/components/UserSelection.js:      Supplier2: { url: 'localhost:8003/graphql', role: 2 },
```

^^ all of the places where `localhost` is set in the UI module must be replaced with your hostname. Also, after making these changes you will need to rebuild the UI container and clear your `localStorage`.

**Comment by MCLDG** [comment]
I've added PR #134 to replace this one. A rebased, simplified version.

This PR can be closed.

@danielnorkin - check the changes I made in PR #134 - the same changes should work for you on Azure.

**Comment by danielnorkin** [comment]
I haven't tested this PR, but I got it working on an Azure Ubuntu VM. I
quickly looked through your readme and some of the files I had to change
and noticed two things:

1. In your readme you may want to add the dependency of installing sol-c.
When buulding the Azure VM version of radish34 I had to install sol-c,
which your new (or the original baseline) readme didnt specificy.

2. The idea of a config file is really nice. I added a PR that includes
comment in the code where manual changes need to be made. I think you are
missing adding a parameter variable in the ServiceWorker file in /ui/src.

**Comment by MCLDG** [comment]
@danielnorkin , it shouldn't be necessary to install sol-c, it's done by the 'make' targets. Thanks for the pointer on ServiceWorker, I'll check it out.

**Comment by MCLDG** [comment]
Replaced by #134, closing this one.

**Pull request closed without merging**